drivers/input/input_stmpe811.c

```diff
@@ -0,0 +1,469 @@
+/**
+ * Copyright (c) 2023 Antmicro <www.antmicro.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#define DT_DRV_COMPAT st_stmpe811
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/input/input.h>
+#include <zephyr/sys/byteorder.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(stmpe811, CONFIG_INPUT_LOG_LEVEL);
+
+#include "input_stmpe811.h"
+
+struct stmpe811_config {
+	struct i2c_dt_spec bus;
+	struct gpio_dt_spec int_gpio;
+	uint16_t screen_width;
+	uint16_t screen_height;
+	uint8_t panel_driver_settling_time;
+	uint8_t touch_detect_delay;
+	uint8_t touch_average_control;
+	uint16_t hold_timeout;
+	uint16_t polling_period;
+	uint16_t release_counter_threshold;
+};
+
+struct stmpe811_data {
+	const struct device *dev;
+	struct k_work work;
+	struct k_work_delayable polling_work;
+	struct gpio_callback int_gpio_cb;
+	uint32_t poll_timeout_us;
+	uint32_t poll_period_us;
+	uint32_t poll_release_threshold;
+	uint32_t poll_release_counter;
+	struct k_thread thread;
+	bool touch_detected;
+	uint32_t touch_x;
+	uint32_t touch_y;
+};
+
+static int stmpe811_reset(const struct device *dev)
+{
+	const struct stmpe811_config *config = dev->config;
+
+	/* Power down the stmpe811 */
+	int ret = i2c_reg_write_byte_dt(&config->bus, STMPE811_SYS_CTRL1_REG, 2);
+
+	if (ret < 0) {
+		return ret;
+	}
+	k_msleep(STMPE811_RESET_DELAY);
+
+	/* Power on the stmpe811 after the power off => all registers are reinitialized */
+	ret = i2c_reg_write_byte_dt(&config->bus, STMPE811_SYS_CTRL1_REG, 0);
+	if (ret < 0) {
+		return ret;
+	}
+	k_msleep(STMPE811_WAIT_DELAY);
+
+	return 0;
+}
+
+static int stmpe811_io_enable_af(const struct device *dev, uint32_t io_pin)
+{
+	const struct stmpe811_config *config = dev->config;
+	uint8_t tmp;
+
+	/* Get the current register value */
+	int r = i2c_reg_read_byte_dt(&config->bus, STMPE811_IO_AF_REG, &tmp);
+
+	if (r < 0) {
+		return r;
+	}
+
+	tmp &= ~io_pin;
+
+	/* Write back the new register value */
+	return i2c_reg_write_byte_dt(&config->bus, STMPE811_IO_AF_REG, tmp);
+}
+
+static uint8_t stmpe811_tsc_config_bits(const struct device *dev)
+{
+	const struct stmpe811_config *config = dev->config;
+	uint8_t ret = 0;
```

Comment:
not much of a point initializing at zero just to |= it once. Either leave it initialized it and assign directly or drop it entirely and return the bits value directly.

Reply:
Done